File: C++/easy/armstrong-numbers/armstrong_numbers.cpp
#include "armstrong_numbers.h"

namespace armstrong_numbers {
  bool is_armstrong_number(const int& num) {
    bool result{false};
    int num_digits{1};
    int aux{num};
    while (aux > 9) {
      aux /= 10;
      ++num_digits;
    }

    aux = num;
    int possible_armstrong_number{0};
    for (int i{0}; i < num_digits; ++i) {
      possible_armstrong_number += pow(aux % 10, num_digits);
      aux /= 10;
    }

    if (possible_armstrong_number == num) result = true;
    return result;
  }
}  // namespace armstrong_numbers

File: Python/easy/armstrong-numbers/armstrong_numbers.py
def is_armstrong_number(number):
    """This functions calculate if a number is an armstrong number

    :param number: The number to calculate whether it is an armstrong number
    :return: True if the number is an armstrong number

    
    """
    possible_armstrong_number = 0
    width = 0
    number_aux = number
    i = number

    while i > 0:
        width += 1
        i //= 10
    
    i = 1
    while i <= width:
        possible_armstrong_number += pow(number_aux % 10, width)
        number_aux //= 10
        i += 1
    
    return number == possible_armstrong_number
